### PR TITLE
Add `Semantic Similarity with KerasNLP` tutorial

### PR DESCRIPTION
**Pull request by shivance**
Issue : https://github.com/keras-team/keras-nlp/issues/726
Kaggle Tutorial [Link](https://www.kaggle.com/code/shivanshuman/kerasnlp-semantic-similarity-with-bert/notebook)

This PR continues the previous PR #1253 

This PR ports Keras-io's existing [Semantic Similarity with BERT](https://keras.io/examples/nlp/semantic_similarity_with_bert/) to KerasNLP. Now the tutorial is pure Keras + tensorflow. It doesn't depend on Tensorflow at all.

### CODE REVIEW

**Comment by shivance** [comment]
@mattdangerw & @fchollet Please review.
I have addressed all the comments from the previous one. The example text turned out a lot better with the help of ChatGPT.

**Comment by shivance** [comment]
> Did you catch this comment from the last PR? [#1253 (comment)](https://github.com/keras-team/keras-io/pull/1253#discussion_r1145564348) We should try to have the triangular rate play out over the entirely of the training run, so I may have messed up my initial suggestion there.

I totally missed it 
image

Will update .

**Comment by shivance** [comment]
@fchollet The PR is ready to merge.
Could you please review?

**Review by fchollet — APPROVED** [review]
Great work! I applied some minor copyedits. Please pull them and generate the md and ipynb files for the tutorial.

**Comment by shivance** [comment]
Hey @fchollet, when I run the autogen script, at later stage it starts to train the model, and my computer keeps crashing there :(

**Comment by fchollet** [comment]
> Hey @fchollet, when I run the autogen script, at later stage it starts to train the model, and my computer keeps crashing there :(

Are you able to produce a rendered notebook by some other means? In that case you could manually edit the `md` and `ipynb` files using your rendered notebook (see what the other files look like for some examples...)

**Comment by mattdangerw** [comment]
@shivance I have added the autogenerated files, but there was one fairly large bug here--the training, validation and test sets were are loading the same data!

I have fixed that, but the hyperparameters are now somewhat off. None of the improvements to the learning rate or optimizer are actually boosting performance. Roberta still boosts things. This is probably something we should look at.

**Comment by mattdangerw** [comment]
OK! Tweaked learning rates and update the commentary, this should be ready to go!

**Comment by fchollet** [comment]
Thank you for the contribution! LGTM
